src/D2L.CodeStyle.Analyzers/ExtensionMethods.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Linq;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+
+namespace D2L.CodeStyle.Analyzers {
+
+	internal static class ExtensionMethods {
+
+		/// <summary>
+		/// Check if the symbol has a specific attribute attached to it.
+		/// </summary>
+		/// <param name="symbol">The symbol to check for an attribute on</param>
+		/// <param name="attributeClassName">The class of the attribute to check for</param>
+		/// <returns>True if the attribute exists on the symbol, false otherwise</returns>
+		public static bool HasAttribute(
+			this ISymbol symbol,
+			string attributeClassName

```

Comment:
Stringly comparisons kind of suck, so I'm a bit unexcited about making this a shared extension method. Checks like these should be done at the symbol level to avoid ambiguity.

Types can be renamed:

```cs
using CheatyType = TypeOurAnalyzerExpects;
```

Types can also be ambiguous (same name in different namespaces, or even the same name+namespace but different assembly)

Reply:
I poked around a bit and saw that we actually do this a lot. Immutability does it, NonNull was doing it...
I think this should be its own issue.